packages/terra-form/tests/jest/NumberField.test.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import NumberField from '../../src/NumberField';
+
+class ControlledInput extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { years_experience: 2 };
+    this.updateYearsExperience = this.updateYearsExperience.bind(this);
+  }
+
+  updateYearsExperience(e) {
+    this.setState({
+      years_experience: e.target.value,
+    });
+  }
+
+  render() {
+    return (
+      <NumberField name="years_experience" value={this.state.years_experience} onChange={this.updateYearsExperience} />
+    );
+  }
+}
+
+it('should render a default component', () => {
+  const input = <NumberField />;
+  const wrapper = shallow(input);
+  expect(wrapper).toMatchSnapshot();
+});
+
+it('should render a NumberField with the rest of the props', () => {
```

Comment:
Correct. defaultValue goes to uncontrolled, and value and onChange do a controlled input